Keeps the blame to the correct code even if code changes has been made

Sertion opened the pull request:
Solves issues #19, #20, and #23.

waderyan commented:
@Sertion these changes look pretty good. Do you want to take this extension over? I don't have time to test the changes and republish. Happy to hand it off if you are interested.

Sertion commented:
@waderyan: If you feel like it would be better of in someone else hands I can give it a shot.

waderyan commented:
Cool let me see what it will take to transfer publishing rights to you

Sertion committed:
2.4.2 - Committer vs Author

Sertion committed:
2.4.3 - Better Show message for unblamed lines

Sertion committed:
2.4.3 - Map over Object for file cache

Sertion committed:
2.4.4 - Updating dependencies

Commit:
Clarify is plural setting

Commit:
Handle ENOENT errors more verbosely

Would want to reject to stop next error, but this should be clear enough

Sertion committed:
Merge pull request #41 from Yottster/master

 Handle ENOENT errors more verbosely

Sertion committed:
Setting strict to true in tsconfig

Sertion committed:
Preparing for 2.5.0. Updated default info message format.

Sertion committed:
Merge branch 'master' of https://github.com/Sertion/vscode-gitblame

Sertion committed:
Stricting execcommand

Sertion committed:
Removing internalHashLength setting

Sertion committed:
Going back to using fs.watcher

Sertion committed:
Fix multiple parallel blames and use vscode.git

Sertion committed:
Updating release date

Sertion committed:
Updating version in package.json

dewe committed:
Create web url from origin without .git suffix

Sertion committed:
Merge pull request #44 from dewe/web-url-without-git-suffix

Create web url from remote url with no .git suffix

Sertion committed:
v2.5.1

Sertion committed:
New parser for defaultWebPath

Also adding new token for commitUrl

Sertion committed:
Additional tests

Sertion committed:
Fix indentation

Sertion committed:
v2.6.0 - More supported git origin formats

Sertion committed:
Merge pull request #46 from Sertion/default-web-path-parsing

New parser for defaultWebPath

Sertion committed:
Add tests for #48

Sertion committed:
Can handle non-alphanumeric characters in git origins

Sertion committed:
Merge pull request #49 from Sertion/issue-48

Dot in origin url is interpreted wrong

Sertion committed:
v2.6.1

Sertion committed:
v2.6.2

allight committed:
Add support for using remote URL in blame link expansion

For some projects one can use the remote-endpoint to create a commit
link. This adds a new token for gitblame.commitUrl '${project.remote}'
that will expand to the current-branches upstream URL with the
protocol, port-numbers, and trailing '.git' stripped off.

Test: Use git-blame with an AOSP tree and gitblame.commitUrl set to
      'https://${project.remote}/+/${hash}'.

allight committed:
Fix lint issues with git remote change

Test: npm run lint

allight committed:
Address comments

The git '--worktree' option is new in git 2.20. Use '--local' instead
since it is older and available on more git installs.

Change await layout.

Test: Use git-blame with an AOSP tree and gitblame.commitUrl set to
      'https://${project.remote}/+/${hash}'.

Sertion committed:
Merge pull request #50 from allight/remote-token

Add support for using remote URL in blame link expansion

Sertion committed:
v2.6.3

Sertion committed:
2.7.0 - closing #47

Sertion committed:
2.7.0 - Bumping version

Sertion committed:
Adding a simple auto plural detector

radar committed:
Show distance of time in words in years

Sertion committed:
Merge pull request #53 from radar/master

Show distance of time in words in years

Sertion committed:
Merge pull request #52 from Sertion/plural-web-path

Adding a simple auto plural detector

Sertion committed:
Generalize plural texts with function

Sertion committed:
Move to ESLint from TSLint

Sertion committed:
Fixing linting in tests

Sertion committed:
Add @typescript-eslint as eslint plugin

Sertion committed:
Fix lint rule config

Sertion committed:
Remove TSLint as a dev dependency

Sertion committed:
Eslinting

Sertion committed:
Move lint before tests

Sertion committed:
Change travis node version to match vscode

Sertion committed:
Updating dependencies

Sertion committed:
Fix status indicator

Sertion committed:
2.8.0

Sertion committed:
2.8.1 - Fixes #54

Sertion committed:
Travis is unable to install node 10.2 for builds

Sertion committed:
Testing the exact preinstalled node version

Sertion committed:
Remove V from node_js version

Sertion committed:
Removing dependency on momentjs

Sertion committed:
More compressed gif

Sertion committed:
Remove custom spinner support

Sertion committed:
Remove the last non-dev dependency

Sertion committed:
Removelog levels

Sertion committed:
Factory is now responsible for workspace knowledge

Sertion committed:
Change cache solution

Sertion committed:
Split out blaming from extension

Sertion committed:
Add TSyringe for IoC

Sertion committed:
Fixes #55 and update test framework

Sertion committed:
New build process

Sertion committed:
Blame the line the tick following the movement

Sertion committed:
Linting

Sertion committed:
Fixing test-runner

Sertion committed:
Add xvfb service

Sertion committed:
Merge pull request #56 from Sertion/slimdown

V3: Slimdown

Sertion committed:
V3.0.1 - Closes #57

Sertion committed:
Starting work on 3.1.0

Sertion committed:
Fix issue with type for info tokens

madprog committed:
Add configuration gitblame.remoteName

This new configuration property allows to specify another remote than
`origin` to build the commit URL.

For example, I like to clone repositories with the remote name
`upstream`, and reserve the name `origin` for the personal repository
to which I can push.

Sertion committed:
Merge pull request #65 from madprog/config-remote-name

Add configuration `gitblame.remoteName`

Sertion committed:
Merge branch 'master' into v3.1.0

abarichello committed:
Format text decorator tests

abarichello committed:
Implement author and commiters 'date' tokens

Sertion committed:
Dependency stuff

Sertion committed:
Fix lint issue

Sertion committed:
Containers when running tests

Sertion committed:
Starting to write tests

Sertion committed:
Proper usage of the git.vscode api

Sertion committed:
Merge branch 'using-git.vscode-api' into view-wrapper

Sertion committed:
Merge pull request #66 from aBARICHELLO/date-token

Implement date token for author and committer

Commit:
Use getGitCommand in test

Commit:
Merge branch 'using-git.vscode-api' into v3.1.0

Sertion committed:
Merge pull request #72 from Yottster/v3.1.0

'using-git.vscode-api' into v3.1.0

Sertion committed:
Write tests for all commands

Sertion committed:
Fix lint

Sertion committed:
Potential fix for sandbox

Sertion committed:
Updated travis file

Sertion committed:
Updating node version

Sertion committed:
Props are not avalible in vscode-test otherwise

Sertion committed:
Merge branch 'v3.1.0' into view-wrapper

Sertion committed:
Merge pull request #75 from Sertion/view-wrapper

View wrapper

Sertion committed:
Closes #58

Sertion committed:
Split up function

Sertion committed:
Separating stream parsing

Sertion committed:
Implements #74

Sertion committed:
cleanup

Sertion committed:
Updating deps

Sertion committed:
Merge pull request #77 from Sertion/view-wrapper

Tests and cleanups

Sertion committed:
Merge pull request #64 from Sertion/v3.1.0

V3.1.0

Sertion committed:
Update required vscode version

Sertion committed:
Moving to Rollup

Sertion committed:
Updated test build

Sertion committed:
Merge pull request #79 from Sertion/rollup

Moving to Rollup

Sertion committed:
Solves issue #78

Sertion committed:
Merge pull request #80 from Sertion/issue-78

Solves issue #78

Sertion committed:
Solves #81

Sertion committed:
Downgrade vscode types

BenLanglois committed:
Fix parameter names in ago.ts

BenLanglois committed:
Remove unused secondsBetween function

BenLanglois committed:
Use smaller units of time when available in toDateText

Sertion committed:
Merge pull request #83 from BenLanglois/improve-time-ago-estimate

Improve time ago estimate

Sertion committed:
Adding the #82 fix

Sertion committed:
Being able to terminate a blame again

Sertion committed:
Updating deps

Sertion committed:
Possible memory improvements

Keep all object structures the same throughout their lifetimes

Sertion committed:
Privated

Sertion committed:
Rewriting token parsing to work better with types

Sertion committed:
Revert "Privated"

This reverts commit a9d7a9fec7cda09d65913a8c211f12b29508e773.

Sertion committed:
Uri.parse strict mode

Sertion committed:
Adding temporary logging

Sertion committed:
No longer allow empty URIs

Sertion committed:
Remove debugg-logging

Sertion committed:
npm audit

Sertion committed:
Trying out github's ci

Sertion committed:
Installing xvfb before node dependencies

Sertion committed:
Set DISPLAY to :0

This is magic and if it works it is pretty neat.

Sertion committed:
Trying out GabrielBB/xvfb-action@v1.2

Sertion committed:
Update blank.yml

Sertion committed:
Delete .travis.yml

Sertion committed:
Update README.md

Sertion committed:
Merge pull request #86 from Sertion/github-ci

Trying out github's ci

Sertion committed:
Merge pull request #85 from Sertion/issue-84

Closes Issue #84

Sertion committed:
V4.1.0

Sertion committed:
No fun allowed

Sertion committed:
v4.2.0

Sertion committed:
Keep port for http(s).

This is a breaking change from previous behavior.

Closes !89

Sertion committed:
Merge pull request #90 from Sertion/issue-89

Keep port for http(s).

Sertion committed:
V.5 - Update vscode require

Sertion committed:
Updating dependencies and tests

Sertion committed:
Refactoring

Generators are cool and all that but I went a bit overboard.
Also removing a few boolean arguments.

Sertion committed:
Updating deps

Sertion committed:
Moving the parsing to its own file

Sertion committed:
Large refactoring. Removing TSyringe.

Trying to get the bundle size down as well as minimize the number
of third party dependencies I ship. It is now down to just vscode.

Sertion committed:
No more blank blames. Empty is the new cool.

Sertion committed:
Commit > Lines and use a buffer instead of large strings

Sertion committed:
More cleanup

Sertion committed:
The big de-Gitening

Sertion committed:
Merge pull request #91 from Sertion/refactor-2020

Refactor 2020

Sertion committed:
V5.0.1

Sertion committed:
V5.0.1 tweak

Sertion committed:
Reduction and fixes #92

Sertion committed:
Merge pull request #93 from Sertion/reduction

Reduction and fixes #92

Sertion committed:
v6.0.1

Sertion committed:
v6.0.2

Sertion committed:
New lint rule and remove inferCommitUrl

Sertion committed:
Final v7

Sertion committed:
Merge pull request #100 from Sertion/version-7

Version 7

Sertion committed:
V7.0.1

Sertion committed:
v7.0.2

Sertion committed:
7.0.3

Sertion committed:
7.0.4

Sertion committed:
Create release.yml

Sertion committed:
Merge pull request #104 from Sertion/release-arts

Create release.yml

Sertion committed:
v7.0.5

Sertion committed:
More tests

Sertion committed:
v7.0.6

Sertion committed:
Merge pull request #105 from Sertion/unknown-issue-with-time

More tests surrounding processing of git and token strings

Sertion committed:
We can now use Node 14

Sertion committed:
We can now use Node 14

Sertion committed:
We can now use Node 14

Sertion committed:
The big constening

Sertion committed:
Add tests for generateUrlTokens

Sertion committed:
Updating tests

Sertion committed:
Fixes #107

Sertion committed:
Fixes #108

Sertion committed:
Resolving weird runner issue

Sertion committed:
Fix issue with test stubs

Sertion committed:
Try to make failed tests fail the action

Sertion committed:
Add test to not regress #110

Sertion committed:
Add specific test for #110

Sertion committed:
Merge pull request #109 from Sertion/es2020

v8.0.0

Sertion committed:
v8.0.1

Fixes #111

Sertion committed:
v8.0.1 Add tslib for Open VSX action

joaomoreno committed:
avoid reloading for settings change

Sertion committed:
Add step for linting

joaomoreno committed:
lint

Sertion committed:
Merge pull request #112 from joaomoreno/joao/avoid-reload

Avoid reloading for settings change

Sertion committed:
v8.1.0

tdhsmith committed:
Add tool token for current line number

Useful for jumping to same position in blame of large files

Sertion committed:
Merge pull request #117 from taylor-literaseed/master

Add token for current line number

Sertion committed:
Fixes #119

angelfraga committed:
feat: allow blaming symlinked files

atkinchris committed:
Remove right-to-left override characters from parsed text

This removes right-to-left override characters from parsed text, to prevent malicious reformatting of status messages.

Sertion committed:
Starting work on 8.2

Fixing test stage in CI

Sertion committed:
Miss in linting

Sertion committed:
Merge pull request #122 from atkinchris/remove-rtlo-from-names

Remove right-to-left override characters from parsed text

Sertion committed:
Merge pull request #121 from angelfraga/master

feat: allow blaming symlinked files

Sertion committed:
Update changelog and remove sync call

Sertion committed:
Update engine

Sertion committed:
v8.2.1 Minor touch-ups

Sertion committed:
Fixes #124

Sertion committed:
The big double quoting

Sertion committed:
Updating changelog 8.2.2

Sertion committed:
Bump package.json versions

azanli committed:
Apply -C option to git blame command for better blame analysis in files changed by reorganizations

Sertion committed:
Move to esbuild

Sertion committed:
v8.2.3

Sertion committed:
Update readme to fix #127. Prepare for version 9

Sertion committed:
Rename job

Sertion committed:
Rename job

Sertion committed:
Rename job

Sertion committed:
Typo

Sertion committed:
Merge pull request #125 from Friendly-Robot/master

Apply `-C` option to git blame for more precise blaming in reorganized files

Sertion committed:
v9 Line info from git

Sertion committed:
Update preview image

Sertion committed:
v9 New default value for gitblame.commitUrl

Sertion committed:
pluralWebPathSubstrings can be undefined

Sertion committed:
Fix #128

Sertion committed:
Update changelog, add test

Sertion committed:
Correct new default value

Sertion committed:
Date for v9 release

Sertion committed:
Don't package esbuild.mjs

Sertion committed:
v9.0.1

f4n0 committed:
WIP: Text Decorations

f4n0 committed:
added options for customization

f4n0 committed:
Pre-PR Validations

f4n0 committed:
Lint + PR Check

f4n0 committed:
Lint Fix

Sertion committed:
Merge pull request #135 from f4n0/master

[FEATURE] Inline Blame View

Sertion committed:
v10

Separate the format for inline and status bar views

Sertion committed:
Update setting names

razinj committed:
Remove inlineBlameColor

Sertion committed:
Merge pull request #136 from razinj/master

Remove inlineBlameColor

Sertion committed:
v10.1

Sertion committed:
checkout@v3

Sertion committed:
Lint

Sertion committed:
Trying without GabrielBB/xvfb-action

Sertion committed:
Got confused by yaml

Sertion committed:
use xvfb-run

aaravind100 committed:
add a delay before blame is shown (#139)

Add a setting to specify a delay before a blame is displayed. This setting is set to 0 by default.

Sertion committed:
Merge pull request #141 from aaravind100/topic/add-delay

add a delay before blame is shown (#139)

Sertion committed:
v10.2.0

Sertion committed:
v10.2.1
